myriad-scheduler/src/main/java/com/ebay/myriad/scheduler/event/handlers/ResourceOffersEventHandler.java

```diff
@@ -105,11 +106,20 @@ public void onEvent(ResourceOffersEvent event, long sequence,
         for (NodeTask nodeTask : nodeTasks) {
           nodeTask.setSlaveAttributes(offer.getAttributesList());
         }
+        // keep this in case SchedulerState gets out of sync. This should not happen with 
+        // synchronizing addNodes method in SchedulerState
+        // but to keep it safe
+        final Set<Protos.TaskID> blacklistedTasks = Sets.newHashSet();
```

Comment:
But the collection we are looping through, ```schedulerState.getPendingTaskIds()```, is different from the collection we want to remove an item from. ```schedulerState.removeTask(taskId)``` is removing an item from ```SchedulerState#tasks```.